### PR TITLE
feat(core): config to limit RSS memory used for writing

### PR DESCRIPTION
**Pull request by ideoma**
Add a new optional configuration setting to limit total RSS memory used by table writer.

Related to https://github.com/questdb/questdb/issues/3931

```
# RSS memory limit used in cairo TableWriter. Attempt to allocate beyond this limit will result in exception
# and table will be suspended. The default is 0 which means no limit.
cairo.writer.memory.limit=14G
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 29 (79.31%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Unsafe.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|

File: core/src/main/resources/io/questdb/site/conf/server.conf [hunk 1]
```diff
@@ -147,6 +147,10 @@ query.timeout.sec=60
 
 ################ Cairo settings ##################
```
Comment (bziobrowski) [inline core/src/main/resources/io/questdb/site/conf/server.conf:149]: Why not set it to 0 to reflect the default value ? 
Are you planning to add he prop to pkg/ami/marketplace/assets/server.conf ?

File: core/src/main/resources/io/questdb/site/conf/server.conf [hunk 2]
```diff
@@ -147,6 +147,10 @@ query.timeout.sec=60
...
 
+# RSS memory limit used in cairo TableWriter. Attempt to allocate beyond this limit will result in exception
+# and table will be suspended. The default is 0 which means no limit.
+#cairo.writer.memory.limit=14G
```
Comment (ideoma) [inline core/src/main/resources/io/questdb/site/conf/server.conf:152]:
```suggestion
#cairo.writer.memory.limit=0
```

File: core/src/main/resources/io/questdb/site/conf/server.conf (hunk 1 above)
Comment (ideoma) [reply core/src/main/resources/io/questdb/site/conf/server.conf:149]: Ok, I set it to 0.
No I don't plan to change AMI config initially

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 29 (79.31%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Unsafe.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**